lib/Models/Terria.ts

```diff
@@ -1006,6 +1013,62 @@ export default class Terria {
           .query("")
           .hash("")
       );
+
+      // /catalog/ and /story/ routes
+
+      if (!document.baseURI.endsWith("/"))
+        console.warn(
+          "Terria expected document.baseURI to end with a '/'. Routes may not work as intended"
+        );
+      if (newUrl.startsWith(document.baseURI)) {
+        const pageUrl = new URL(newUrl);
+        // Find relative path from baseURI to documentURI excluding query and hash
+        // then split into url segments
+        // e.g. "http://ci.terria.io/main/story/1#map=2d" -> ["story", "1"]
+        const segments = (pageUrl.origin + pageUrl.pathname)
+          .slice(document.baseURI.length)
+          .split("/");
+        if (
+          segments.length === 2 &&
+          segments[0] === "catalog" &&
+          segments[1].length > 0
+        ) {
+          this.initSources.push({
+            name: `Go to ${pageUrl.pathname}`,
+            errorSeverity: TerriaErrorSeverity.Error,
+            data: {
+              previewedItemId: decodeURIComponent(segments[1])
+            }
+          });
+          const replaceUrl = new URL(newUrl);
+          replaceUrl.pathname = new URL(document.baseURI).pathname;
+          history.replaceState({}, "", replaceUrl.href);
+        } else if (
+          segments.length === 2 &&
+          segments[0] === "story" &&
+          segments[1].length > 0 &&
+          isDefined(this.configParameters.storyRouteUrlPrefix)
+        ) {
+          let storyJson;
+          try {
+            storyJson = await loadJson(
+              `${this.configParameters.storyRouteUrlPrefix}${encodeURIComponent(
+                this.appName
+              )}/${segments[1]}`
```

Comment:
Also, should we use `appName` as part of the URL? It means that e.g. nationalmap and DEA can share the same stories bucket and have slightly different stories for the same story id (which I think is essential, I don't think we can't get away with globally unique story ids). The alternative is to put that in the `storyRouteUrlPrefix`, which may be a better idea.

Reply:
For whether to use `appName`. I think I prefer the second option of changing the `storyRouteUrlPrefix` per map because it is less confusing and avoids having to explicitly communicate everywhere that the share json URL has to be `storyRouteUrlPrefix + appName`.

Reply:
For proxying the story route url, the could configure it as:

`"storyRouteUrlPrefix": "proxy/https://tile-test.terria.io/stories/"` and add it to the proxy list.

So, shouldn't be a problem, I guess.

Reply:
One other option is to make `storyRouteUrlPrefix` a template string that'll accept `{appName}`

Reply:
I agree @na9da. I've changed the behaviour to leave out appName from URL construction. At the moment I won't implement storyRouteUrlPrefix as a template string, but I'll keep it in mind if requirements change.